### PR TITLE
redesigned annotators

### PR DESCRIPTION
**Pull request by SkalskiP**
This PR expands our range of Annotators and makes them more reusable.

bounding-box-annotator-example
box-corner-annotator-example
ellipse-annotator-example
label-annotator-example-2
mask-annotator-example

### CODE REVIEW

**Comment by SkalskiP** [comment]
@hardikdava I created a feature branch that we can use to redesign annotators and make them more modular according to our discussion here https://github.com/roboflow/supervision/issues/168

I decided that we will conduct all work around annotators on this feature branch and when we are done we will merge this branch as a whole into `main`.

**Comment by hardikdava** [comment]
Hi @SkalskiP 👋 , cool!. I will add `modified box annotators`, `trace` and `trace-annotators` in this branch.

**Comment by hardikdava** [comment]
@SkalskiP I have created initial PR as #171 . Let me know how can I contribute in this week.

**Comment by hardikdava** [comment]
Test script to test new annotators.

```
import supervision as sv
import cv2
import numpy as np
from ultralytics import YOLO

img = cv2.imread("../data/zidane.jpg")

model = YOLO("yolov8s-seg.pt")

yolo_detect_result = model.predict([img], conf=0.5)[0]
detections = sv.Detections.from_yolov8(yolo_detect_result)

annotated_img = img.copy()

cat_maps = {0: "person", 27: "tie"}
labels = [f"{cat_maps[class_id]}" for _, _, _, class_id, _ in detections]

tracker_ids = np.array([1, 2, 3], dtype=int)
detections.tracker_id = tracker_ids  # Add tracker id to visualize color by tracker id

detection_annotator = sv.DetectionAnnotator()
annotated_img = detection_annotator.annotate(scene=annotated_img, detections=detections, labels=labels)

# segmentation_annotator = sv.SegmentationAnnotator()
# annotated_img = segmentation_annotator.annotate(scene=annotated_img, detections=detections, labels=labels)

# label_annotator = sv.LabelAnnotator()
# annotated_img = label_annotator.annotate(scene=annotated_img, detections=detections, labels=labels, color_by_track=True)

# mask_annotator = sv.MaskAnnotator()
# annotated_img = mask_annotator.annotate(scene=annotated_img, detections=detections, color_by_track=True)

# box_annotator = sv.BoxAnnotator()
# annotated_img = box_annotator.annotate(scene=annotated_img, detections=detections, color_by_track=True)

cv2.imshow("Result", annotated_img)
cv2.waitKey(0)
```

**Comment by hardikdava** [comment]
@SkalskiP I made progress on this feature. `PillowLabelAnnotator` is only not implemented yet. All other annotators are implemented and tested locally.

**Comment by hardikdava** [comment]
Hi @SkalskiP :wave: I am planning to keep continue contribute to `supervision`. I have some questions :confused:,

- How to update docs?
- Should we start using Github Projects? That will be useful for planning, prioritization, task assignment and keeping tracks on the issue.

**Comment by SkalskiP** [comment]
@hardikdava, how about meeting for 30 minutes on video call and talking through the priorities? 

In principle, I would like us to be focused on those tasks in the first place: https://github.com/roboflow/supervision/discussions/181, but with all the loose ends I think meeting is the right move.

**Comment by hardikdava** [comment]
@SkalskiP Of course, we can talk. Let me know when we can have a short call.

**Comment by SkalskiP** [comment]
@hardikdava zoom or google meet?

**Comment by hardikdava** [comment]
Google meet is fine.

**Comment by SkalskiP** [comment]
@hardikdava you are based in Germany? Can we meet at 8 PM? Or you'd like to do it earlier?

**Comment by hardikdava** [comment]
@SkalskiP Yeah, I am based in Germany. Would it possible to talk at 6 PM?

**Comment by SkalskiP** [comment]
@hardikdava, not today, but I could meet at 6 PM tomorrow. How about that?

**Comment by hardikdava** [comment]
@SkalskiP let's talk at 8 PM today then. I can make it.

**Comment by hardikdava** [comment]
@SkalskiP On side note, I am adding `PillowLabelAnnotator()`. Do we have any specific default font?

**Comment by SkalskiP** [comment]
@hardikdava can we move the meeting to tomorrow? I still have a lot of work today :/ And I won't make it :/ I'm super sorry.

**Comment by hardikdava** [comment]
> @hardikdava can we move the meeting to tomorrow? I still have a lot of work today :/ And I won't make it :/ I'm super sorry.

Okay. Let's move it to tomorrow then. Can you suggest free time? I am available after 6 PM (CET) tomorrow.

**Comment by SkalskiP** [comment]
@hardikdava I can do it 6 PM 🔥

**Comment by SkalskiP** [comment]
Hi, @hardikdava 👋🏻! We wanted to release this within `supervision-0.12.0`, but I don't have enough time to review it today. I plan to publish `supervision-0.12.0` tomorrow around 11.00 AM. I need to roll it over to `supervision-0.13.0`. I'm sorry. It's my fault.

**Comment by hardikdava** [comment]
@SkalskiP , No worries. Then we can release every annotators together. Please start review other PRs or let me know if you need for reviewing them. Always happy to help!

**Comment by SkalskiP** [comment]
It looks like we are a bit blocked here. I reached out to `dong.li` on LinkedIn. I hope he accepts and we will be able to push forward.

**Comment by Ucag** [comment]
Sorry for my block here, I've accepted the CLA.

**Comment by SkalskiP** [comment]
Thanks, @Ucag 👋🏻! Looks like CLAassistant cant tie `dong.li` user with your user.

**Comment by Ucag** [comment]
> User name for my github is just a nickname. Is there anything else I should do to help push everything forward? My work email differs from my personal email as well as `user.name` in my local git config.

**Comment by SkalskiP** [comment]
I'm afraid not @Ucag. But don't worry. We will figure it out.

**Comment by hardikdava** [comment]
@onuralpszr can you help us testing this feature?
@SkalskiP let's finish this long standing issue.

**Comment by onuralpszr** [comment]
@hardikdava  I am starting to test and solving conflicts and enable tests so we can see everything else as well.

File: setup.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ def get_version():
         'numpy>=1.20.0',
         'opencv-python',
         'matplotlib',
-        'pyyaml'
+        'pyyaml',
+        'pillow'
```
Comment (hardikdava) [inline setup.py:31]: @onuralpszr do I have to specify `pillow` version here? or `pyproject` can handle this by itself?

Comment (onuralpszr) [reply setup.py:31]: Setup.py is gone, I was working but I got fallen to slept last night, give me bit more time. I will handle it.  :)

Comment (hardikdava) [reply setup.py:31]: I know setup.py is gone. This was created long before that. Take your time.

Comment (onuralpszr) [reply setup.py]: > @onuralpszr do I have to specify `pillow` version here? or `pyproject` can handle this by itself?

pyproject will handle it.

**Comment by onuralpszr** [comment]
Video result from tracking example with updated code style.

https://drive.google.com/file/d/1XnChhFs4OFaEMn7HjD6wheJFUve0NZrc/view?usp=sharing

PS : I did not add EllipseAnnotator into example but added video for testing purposes

**Comment by onuralpszr** [comment]
cc @hardikdava  I updated the branch did merge also fix pre-commit 
After those I started with new Annotators on example we have and modified it. Can you check until the current changes. I also did some changes on docstring/remove unused param and remove "old" annotators from '`__init__`.py' because of conflict.  Btw I really wanted to add "Circle" style I already made video and I can adapt to here easily ?

**Review by onuralpszr — COMMENTED** [review]
@hardikdava I updated the branch did merge also fix pre-commit
After those I started with new Annotators on example we have and modified it. Can you check until the current changes. I also did some changes on docstring/remove unused param and remove "old" annotators from '__init__.py' because of conflict. Btw I really wanted to add "Circle" style I already made video and I can adapt to here easily, what do you think about it ?

**Comment by onuralpszr** [comment]
cc @SkalskiP

**Comment by onuralpszr** [comment]
@SkalskiP  we really really need to merge this, that's why I waited for circle part as well.

**Comment by SkalskiP** [comment]
@onuralpszr I'm on it :)

**Comment by onuralpszr** [comment]
@SkalskiP  after you finish may I add circle into here or separate PR ? (either way I have to rebase and code change here would be quicker If you okay with it)

**Comment by SkalskiP** [comment]
@onuralpszr sure you can merge it in once I'm ready. I'm still not done, but tests are looking good. You can take a look here: 

https://colab.research.google.com/drive/1iso4sYNNERe6KqH6zUixF7XwsapbCgzZ?usp=sharing

**Comment by SkalskiP** [comment]
@hardikdava and @onuralpszr take a look at the PR 🙏🏻 

I dropped a few things from PR. We will add them to the next PRs. Still, we are getting most of the functionalities here. 

I tested the changes here: https://colab.research.google.com/drive/1iso4sYNNERe6KqH6zUixF7XwsapbCgzZ?usp=sharing

**Review by hardikdava — APPROVED** [review]
Everything looks fine to me.

**Comment by onuralpszr** [comment]
I am still looking and adding something will post soon

**Comment by SkalskiP** [comment]
@onuralpszr let me know when you are ready ;)

**Comment by SkalskiP** [comment]
@onuralpszr, are we ready to merge?

**Review by onuralpszr — APPROVED** [review]
@hardikdava  @SkalskiP  thank you both for your awesome work. I am merging into develop let's continue make supervision even better.

**Pull request merged**